hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/schema/SchemaReplicator.java

```diff
@@ -168,12 +171,21 @@ private boolean isSchemaReplicated(long schemaId) {
      *
      * @param schemas to replicate.
      */
-    public CompletableFuture<Void> replicateAll(List<Schema> schemas) {
-        CompletableFuture[] replications = schemas.stream()
+    public InternalCompletableFuture<Void> replicateAll(List<Schema> schemas) {
+        InternalCompletableFuture[] replications = schemas.stream()
                 .map(this::replicate)
-                .toArray(CompletableFuture[]::new);
-
-        return CompletableFuture.allOf(replications);
+                .toArray(InternalCompletableFuture[]::new);
+
+        InternalCompletableFuture<Void> future = new InternalCompletableFuture<>();
+        CompletableFuture.allOf(replications)
```

Comment:
which thread is used for `CompletableFuture.allOf` method?

Reply:
What do you mean by that?